### PR TITLE
new: Add spec file for ServicePublication

### PR DESCRIPTION
**Pull request by malkhamis-panw**
This PR adds a new spec file and exposes route `/servicepublication`. This endpoint will be used to make existing services available in sibling namespaces. It will be up to the client to properly construct the service field in payload.

### CODE REVIEW

File: specs/servicepublication.spec [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Model
+model:
+  rest_name: servicepublication
+  resource_name: servicepublications
+  entity_name: ServicePublication
+  package: squall
+  group: core/service
+  description: TODO.
+  update:
```
Comment (primalmotion) [inline specs/servicepublication.spec:9]: you don't need this

File: specs/root.spec [hunk 2]
```diff
@@ -981,6 +981,10 @@ relations:
   create:
     description: Creates a new service dependency.
 
+- rest_name: servicepublication
+  create:
+    description: TODO.
```
Comment (primalmotion) [inline specs/root.spec:986]: please write descriptions immediately. otherwise the TODO will end up in prod, that's almost a gurantee

**Review by primalmotion — CHANGES_REQUESTED** [review]

**Comment by malkhamis-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "service-publication",
    "component": "gaia",
    "pr-id": "662",
    "commit-sha": "0bdd3d0a5966276c7ad2df36765bbc644312c459",
    "pipeline": "master"
  },
  {
    "project": "service-publication",
    "component": "backend",
    "pr-id": "1210",
    "commit-sha": "1af7109b6114b4166a6df1f8eda38a62c73f5de8",
    "pipeline": "master"
  }
]
```

**Comment by malkhamis-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "service-publication",
    "component": "backend",
    "pr-id": "1210",
    "commit-sha": "1af7109b6114b4166a6df1f8eda38a62c73f5de8",
    "pipeline": "master"
  },
  {
    "project": "service-publication",
    "component": "gaia",
    "pr-id": "662",
    "commit-sha": "0bdd3d0a5966276c7ad2df36765bbc644312c459",
    "pipeline": "master"
  }
]
```

**Comment by malkhamis-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "service-publication",
    "component": "backend",
    "pr-id": "1210",
    "commit-sha": "a85a71577b25ec77f4b492bffab322af0a9a5638",
    "pipeline": "master"
  },
  {
    "project": "service-publication",
    "component": "gaia",
    "pr-id": "662",
    "commit-sha": "cde1351bbe12a73f7ae059906a1fbe28ccf20757",
    "pipeline": "master"
  }
]
```

**Comment by malkhamis-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "service-publication",
    "component": "backend",
    "pr-id": "1210",
    "commit-sha": "a85a71577b25ec77f4b492bffab322af0a9a5638",
    "pipeline": "master"
  },
  {
    "project": "service-publication",
    "component": "gaia",
    "pr-id": "662",
    "commit-sha": "cde1351bbe12a73f7ae059906a1fbe28ccf20757",
    "pipeline": "master"
  }
]
```

**Comment by malkhamis-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "service-publication",
    "component": "backend",
    "pr-id": "1210",
    "commit-sha": "dced1f121d86488069e86b94d4a40bebb16b3f93",
    "pipeline": "master"
  },
  {
    "project": "service-publication",
    "component": "gaia",
    "pr-id": "662",
    "commit-sha": "9433c623fd637567bece1175d60fc680125f2ac3",
    "pipeline": "master"
  }
]
```

File: specs/servicepublication.spec [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    Encapsulates a service object that is ought to be published so it can be used
+    in a sibling namespace.
+
+# Attributes
+attributes:
+  v1:
+  - name: service
+    description: The service object that will be published.
+    type: ref
+    exposed: true
```
Comment (primalmotion) [inline specs/servicepublication.spec:18]: This property should be required

**Comment by malkhamis-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "service-publication",
    "component": "backend",
    "pr-id": "1210",
    "commit-sha": "2f05c41ecf51918a1c5e9b45545a941df219ca1a",
    "pipeline": "master"
  },
  {
    "project": "service-publication",
    "component": "gaia",
    "pr-id": "662",
    "commit-sha": "cb0274a92f7f823608006404f02a9bd70dcff000",
    "pipeline": "master"
  }
]
```

**Review by primalmotion — APPROVED** [review]

**Comment by malkhamis-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "service-publication",
    "component": "backend",
    "pr-id": "1210",
    "commit-sha": "3fc0957650287b10e20abdaa14462bc2a4972ea4",
    "pipeline": "master"
  },
  {
    "project": "service-publication",
    "component": "gaia",
    "pr-id": "662",
    "commit-sha": "b62918f4d85ad8c3390174259409dab69773b934",
    "pipeline": "master"
  }
]
```

**Comment by malkhamis-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "service-publication",
    "component": "gaia",
    "pr-id": "662",
    "commit-sha": "b62918f4d85ad8c3390174259409dab69773b934",
    "pipeline": "master"
  },
  {
    "project": "service-publication",
    "component": "backend",
    "pr-id": "1210",
    "commit-sha": "ed6abdf08753396abe67659eef4672c0f5c55636",
    "pipeline": "master"
  }
]
```

**Comment by malkhamis-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "service-publication",
    "component": "backend",
    "pr-id": "1210",
    "commit-sha": "571be1d8f4fe9e51238ff8200e895605f7d5cc89",
    "pipeline": "master"
  },
  {
    "project": "service-publication",
    "component": "gaia",
    "pr-id": "662",
    "commit-sha": "b62918f4d85ad8c3390174259409dab69773b934",
    "pipeline": "master"
  }
]
```

**Comment by malkhamis-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "service-publication",
    "component": "backend",
    "pr-id": "1210",
    "commit-sha": "2ec89d3499e2a8601f32f1d03e0f008f11ac8a0b",
    "pipeline": "master"
  },
  {
    "project": "service-publication",
    "component": "gaia",
    "pr-id": "662",
    "commit-sha": "b62918f4d85ad8c3390174259409dab69773b934",
    "pipeline": "master"
  }
]
```

**Pull request merged**